src/MainWindow.h

```diff
@@ -0,0 +1,257 @@
+/*##################################################################################
+#   File name:          MainWindow.h
+#   Created on:         08/30/2024
+#   Author:             stwomey
+#   Description:        Header file for the MainWindow
+#   Change History:     See https://github.com/SeanTwomey9/PersonalFinanceTool
+#   Relative Location:  PersonalFinanceTool/src/
+##################################################################################*/
+
+#ifndef MAINWINDOW_H
+#define MAINWINDOW_H
+
+#include "Bill.h"
+#include "BillWidget.h"
+
+#include <QMainWindow>
+#include <QSettings>
+#include <QTableWidget>
+
+/**
+ * @brief The MainWindow class represents the primary window which will display the user's personal financial information.
+ */
+class MainWindow : public QMainWindow
+{
+    Q_OBJECT
+
+public:
+
+    /**
+     * @brief Constructs the MainWindow by initializing various UI components and kicking off the initial processing of the user's financial information.
+     */
+    MainWindow();
+
+    /**
+     * @brief Destructs the MainWindow by appropriately deleting objects which were dynamically allocated.
+     */
+    ~MainWindow();
+
+signals:
+
+    /**
+     * @brief Emitted when the application needs to be terminated.
+     */
+    void conditionToTerminateMet();
+
+public slots:
+
+    /**
+     * @brief Terminates the application.
+     */
+    void terminateApplication();
+
+    /**
+     * @brief Attempts to open the config file, if successful a new Bill object is created for the current information inputted into the BillWidget.
+     * Information is also written as a new bill entry in the config file.
+     */
+    void openConfigForBillCreation();
+
+    /**
+     * @brief Calls the openConfigForBillCreation() helper method, before the BillWidget is reset for entry of another bill.
+     */
+    void saveBillAndDisplayBillWidget();
+
+    /**
+     * @brief Calls the openConfigForBillCreation() helper method, before the BillWidget is hidden.
+     * The bill map is then iterated over, and its contents are used to display financial information to the user.
+     */
+    void saveBillAndDisplayDashboard();
+
+    /**
+     * @brief Called when the Save button is pressed. Opens the config file and initially updates the bill map to match the contents of the bill table widget.
```

Comment:
Either refer to widgets in normal case (like this) or use pascal case (*BillTableWidget*).

Reply:
Since the "bill table widget" is not an official class I created, it's appropriate to refer it to this way as opposed to using pascal case like I do with the *BillWidget*.